File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/VerificationStatusAdapter.java
package com.workforcex.worker.ui.worker;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.workforcex.worker.R;
import com.workforcex.worker.api.Verification;
import java.util.List;

public class VerificationStatusAdapter extends RecyclerView.Adapter<VerificationStatusAdapter.ViewHolder> {

    private final List<Verification> verifications;

    public VerificationStatusAdapter(List<Verification> verifications) {
        this.verifications = verifications;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_verification_status, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Verification verification = verifications.get(position);
        holder.tvType.setText(verification.verificationType);
        holder.tvStatus.setText(verification.status);
    }

    @Override
    public int getItemCount() {
        return verifications.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvType, tvStatus;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvType = itemView.findViewById(R.id.tvVerificationType);
            tvStatus = itemView.findViewById(R.id.tvVerificationStatus);
        }
    }
}

File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/VerificationStatusAdapter.kt
package com.workforcex.worker.ui.worker

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.workforcex.worker.R
import com.workforcex.shared.models.Verification

class VerificationStatusAdapter(private val verifications: List<Verification>) :
    RecyclerView.Adapter<VerificationStatusAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_verification_status, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val verification = verifications[position]
        holder.tvType.text = verification.verificationType
        holder.tvStatus.text = verification.status
    }

    override fun getItemCount(): Int {
        return verifications.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvType: TextView = itemView.findViewById(R.id.tvVerificationType)
        val tvStatus: TextView = itemView.findViewById(R.id.tvVerificationStatus)
    }
}